src/iceberg/transform.cc

```diff
@@ -306,6 +306,66 @@ Result<std::unique_ptr<UnboundPredicate>> Transform::Project(
   std::unreachable();
 }
 
+Result<std::unique_ptr<UnboundPredicate>> Transform::ProjectStrict(
```

Comment:
`Transform::ProjectStrict` looks so similar to `Transform::Project` with the only difference being the calls to `ProjectionUtil`. Perhaps we can use template to dispatch it?